src/text-field-base.tsx

```diff
@@ -1,19 +1,67 @@
 'use client';
 import * as React from 'react';
 import {Label, HelperText, FieldContainer} from './text-field-components';
-import {LABEL_LEFT_POSITION, LABEL_SCALE_MOBILE, LABEL_SCALE_DESKTOP} from './text-field-components.css';
+import {LABEL_SCALE_MOBILE, LABEL_SCALE_DESKTOP} from './text-field-components.css';
 import {Text3} from './text';
-import {isRunningAcceptanceTest, isFirefox, isSafari} from './utils/platform';
+import {isRunningAcceptanceTest, isFirefox} from './utils/platform';
 import {useAriaId, useTheme, useScreenSize, useIsomorphicLayoutEffect} from './hooks';
 import classNames from 'classnames';
 import {combineRefs} from './utils/common';
 import * as styles from './text-field-base.css';
-import {sprinkles} from './sprinkles.css';
 import {vars} from './skins/skin-contract.css';
+import {BaseIconButton} from './icon-button';
+import {ThemeVariant} from './theme-variant-context';
 
+import type {DataAttributes, IconProps} from './utils/types';
 import type {InputState} from './text-field-components';
 import type {FieldValidator} from './form-context';
 
+const isValidInputValue = (value?: string, inputType?: React.HTMLInputTypeAttribute) => {
+    if (!inputType) {
+        return true;
+    }
+
+    const input = document.createElement('input');
+    input.type = inputType;
+    input.value = value || '';
+    return input.value !== '';
+};
```

Comment:
Should empty string be considered as valid?

```suggestion
const isValidInputValue = (value: string = '', inputType?: React.HTMLInputTypeAttribute) => {
    if (!inputType) {
        return true;
    }

    const input = document.createElement('input');
    input.type = inputType;
    input.value = value;
    return input.value === value;
};
```

Reply:
No. This is for date inputs, and empty strings are not valid there.